### ISSUE TITLE
[NEXT] The --video-quality flag has sense only for the libx264 video encoder

### ISSUE THREAD

**Issue by pozitronik** [s0md3v/roop#423]
There is a `--video-quality` flag (in the `next` branch) which sets the `-crf` parameter for ffmpeg. However, this parameter is only applicable to the `libx264` encoder.
If `--video-encoder` is set and different from `libx264` there may be another parameters to adjust output video quality.
For the `h264_nvenc` encoder the `-cq` or `-qp` parameters should be used.
The `h264_amf` encoder has `-qp_i` and `-qp_p` and `-qp_b` parameters (yes, they are different and can be used together)
The `h264_videotoolbox` output quality can be configured via desired bitrate parameter: `-b:v`
The `h264_qsv` uses a `-global_quality` parameter.

And so on.

Perhaps it will be the right decision to allow users to set the ffmpeg parameters themselves, like:
`--ffmpeg-params='--video-encoder h264_videotoolbox -b:v 4M'`

**Comment by henryruhs** [comment]
Hmm, I would rather remove the feature again than open this box.

**Comment by henryruhs** [comment]
Fixed via https://github.com/s0md3v/roop/commit/2beb6c23e64f79f1237cb584f4a28a1d5ae69e17

**Comment by pozitronik** [comment]
I see some strange decisions here.
First, you limit users from selecting their preferred encoder, which directly contradicts what you said: "[Let the user decide what encoder to use.](https://github.com/s0md3v/roop/issues/376#issuecomment-1581280834)"
Second, there is no encoder named `nvenc` in FFmpeg. If you try to use the `--video-encoder=nvenc` parameter, you won't get any results. The correct encoder names are `h264_nvenc`, `av1_nvenc`, and `hevc_nvenc`. It's important to use the correct encoder names when working with FFmpeg.

**Comment by C0untFloyd** [comment]
Not really about this issue but I'm wondering why doesn't roop use the movie.py ffmpeg wrapper? I've seen it used in most other face swapping apps and should make it easier for users to use it.

**Comment by phineas-pta** [comment]
@C0untFloyd there is a discussion at #403

TLDR is that almost all wrappers use `subprocess` (except `pyav`), for now ffmpeg commands are still simple, no need for additional dependencies
